Cpp_test/lib/ipc-handler.cpp

```diff
@@ -2,90 +2,150 @@
 #include <ctime>
 #include <iostream>
 
-
-void ipcHandler::select_options(IPCMode mode, int argc, char *argv[]){
+void ipcHandler::select_options(IPCMode mode, int argc, char *argv[])
+{
 	this->info.mode = mode;
-  	int opt;
-	bool fflag = false;
-	bool qflag = false;
-	bool pflag = false;
-	bool sflag = false;
-  	bool hflag = false;
-
+	int opt;
+	int fflag = 0;
+	int qflag = 0;
+	int pflag = 0;
+	int sflag = 0;
+	int hflag = 0;
+
+	if (argc > 5)
+	{
+		throw std::runtime_error("ERROR: Too many arguments, use -h for more information.\n");
+	}
 
-	while ((opt = getopt(argc, argv, "hf:q:p:s:")) != -1) {
-		switch (opt) {
-			case 'h':
-				info.protocol = IPCProtocol::HELP;	
-        		hflag = true;
-				std::cout << "<HELP>\n"
-					"-q /<queue_name>, use queue as IPC method\n"
-					"-p <pipe_name>, uses pipes as IPC method\n"
-					"-s <shared_name> uses shared memory as IPC method\n"
-					"-f <filename>, file used to read/write data\n"
-					"</HELP>\n";
+	while ((opt = getopt(argc, argv, "hf:q:p:s:")) != -1)
+	{
+		switch (opt)
+		{
+		case 'h':
+			info.protocol = IPCProtocol::HELP;
+			hflag = true;
```

Comment:
hflag is used both as an boolean and as an integer
```suggestion
			hflag = 1;
```

Reply:
Fixed.